pymc_extras/distributions/discrete.py

```diff
@@ -399,3 +400,127 @@ def dist(cls, mu1, mu2, **kwargs):
             class_name="Skellam",
             **kwargs,
         )
+
+
+class ShiftedBetaGeometricRV(RandomVariable):
+    name = "sbg"
+    signature = "(),()->()"
+
+    dtype = "int64"
+    _print_name = ("ShiftedBetaGeometric", "\\operatorname{ShiftedBetaGeometric}")
+
+    @classmethod
+    def rng_fn(cls, rng, alpha, beta, size):
+        if size is None:
+            size = np.broadcast_shapes(alpha.shape, beta.shape)
+
+        alpha = np.broadcast_to(alpha, size)
+        beta = np.broadcast_to(beta, size)
+
+        p = rng.beta(a=alpha, b=beta, size=size)
+
+        samples = rng.geometric(p, size=size)
+
+        return samples
```

Comment:
Use a SymbolicRandomVariable that defines the random method in terms of pytensor operations. That way it will work automatically in all backends (jax/numba), not just C/python

Reply:
Here is an example: https://github.com/pymc-devs/pymc/blob/9f653a6ebcc4a205b1f506e2f3e513753d718b28/pymc/distributions/discrete.py#L393-L412